_posts/2021-06-18-a-tour-of-artsys-web-front-end.markdown

```diff
@@ -0,0 +1,53 @@
+---
+layout: epic
+title: "A Tour Of Artsy's Web Front-end"
+date: 2021-06-18
+categories: [architecture]
+author: steve-hicks
+comment_id: 690
+---
+
+- we have articles that cover individual libraries/tools we use in our web front-end
+  - but we don't have a single place that lists them all
+  - this article is that.
+
+What happens when you make a request to artsy.net/some-url?
+
+- [express server](https://github.com/artsy/force/blob/89ce00df9816e892f456e6885fab17c9ab539235/src/v2/server.ts#L52)
+  - Built with [TypeScript])(https://artsy.github.io/blog/2019/04/05/omakase-typescript/)
+  - uses the [found](https://github.com/4Catalyzer/found) router
+    - why do we use found???
+    - each app defines
+      [its own routes & they're aggregated](https://github.com/artsy/force/blob/89ce00df9816e892f456e6885fab17c9ab539235/src/v2/routes.tsx#L35)
+- renders a react app server-side
+  - https://github.com/artsy/force/blob/89ce00df9816e892f456e6885fab17c9ab539235/src/v2/Artsy/Router/buildServerApp.tsx#L164
+  - [`Boot` is our React wrapper](https://github.com/artsy/force/blob/89ce00df9816e892f456e6885fab17c9ab539235/src/v2/Artsy/Router/Boot.tsx#L39)
```

Comment:
We've defined enough of our own custom tooling around SSR routing / bundle splitting / etc that I think we could describe it as "Artsy's server-side rendered, single page app routing framework": https://github.com/artsy/force/tree/master/src/v2/Artsy/Router. It happens to use found (just as it happens to use a few other libs) and we can talk about how they fit together.